server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustLanguagePlugin.kt

```diff
@@ -0,0 +1,175 @@
+package org.jetbrains.bsp.bazel.server.sync.languages.rust
+
+import ch.epfl.scala.bsp4j.BuildTarget
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+import ch.epfl.scala.bsp4j.RustToolchain
+import ch.epfl.scala.bsp4j.RustToolchainResult
+import ch.epfl.scala.bsp4j.RustWorkspaceResult
+import ch.epfl.scala.bsp4j.RustcInfo
+import org.jetbrains.bsp.bazel.info.BspTargetInfo
+import org.jetbrains.bsp.bazel.info.BspTargetInfo.RustCrateInfo
+import org.jetbrains.bsp.bazel.info.BspTargetInfo.TargetInfo
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver
+import org.jetbrains.bsp.bazel.server.sync.languages.LanguagePlugin
+import org.jetbrains.bsp.bazel.server.sync.model.Label
+import org.jetbrains.bsp.bazel.server.sync.model.Language
+import org.jetbrains.bsp.bazel.server.sync.model.Module
+
+class RustLanguagePlugin(private val bazelPathsResolver: BazelPathsResolver) : LanguagePlugin<RustModule>() {
+
+    private val rustWorkspaceResolver = RustWorkspaceResolver(bazelPathsResolver)
+
+    private fun TargetInfo.getRustCrateInfoOrNull(): RustCrateInfo? =
+        this.takeIf(TargetInfo::hasRustCrateInfo)?.rustCrateInfo
+
+    override fun applyModuleData(moduleData: RustModule, buildTarget: BuildTarget) {
+        // TODO
```

Comment:
???

Reply:
Since there is no `RustBuildTarget` in Rust extension there is additional data to apply. In pushed version, this TODO is removed